bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/LocalFolderWatcherHandler.java

```diff
@@ -109,7 +108,7 @@ public void dispose() {
     private void refreshFolderInformation() {
         final String rootDir = config.localDir;
         try {
-            currentLocalListing.clear();
+            List<String> currentLocalListing = new ArrayList<>();
```

Comment:
same here

Reply:
Yes, this is only use in the current function, so i removed global declaration and will only keep it locally